# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: edba72c0c8c6
# Full Hash: edba72c0c8c6ca88aa62634931c16b247a7e4a11
# Author: Brendan Dahl <bdahl@mozilla.com>
# Date: 2019-05-23 04:37:46
# Regressor Bug: 1551320
# File Overlap Count: 1
# Description:
#   Bug 1551320 - Don't allow CreateElement in XUL documents. r=bzbarsky
#   
#   Soon XUL documents will be loaded as XHTML and createElement will create HTML
#   elements instead of XUL element by default. This restriction can be removed
#   once we migrate everything.
# ==============================================================================

diff -r ce28e766055d -r edba72c0c8c6 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Wed May 22 14:56:48 2019 +0000
+++ b/dom/base/Document.cpp	Wed May 22 03:02:53 2019 +0000
@@ -5523,6 +5523,17 @@
     return nullptr;
   }
 
+  // Temporary check until XULDocument has been removed.
+  if (IsXULDocument()) {
+#if DEBUG
+    xpc_DumpJSStack(true, true, false);
+#endif
+    MOZ_DIAGNOSTIC_ASSERT(false,
+                          "CreateElement() not allowed in XUL document.");
+    rv.Throw(NS_ERROR_FAILURE);
+    return nullptr;
+  }
+
   bool needsLowercase = IsHTMLDocument() && !IsLowercaseASCII(aTagName);
   nsAutoString lcTagName;
   if (needsLowercase) {
